### PR TITLE
Prefix/test also all ID fields

### PR DESCRIPTION
**Pull request by mvorisek**

### CODE REVIEW

**Review by georgehristov — APPROVED** [review]

File: demos/collection/lister-ipp.php [hunk 1]
```diff
@@ -50,7 +50,7 @@
     $row->iso = mb_strtolower($row->iso);
 });
 $lister->setModel(new Country($app->db))
-    ->addCondition('id', -1); // no such records so model will be empty
+    ->addCondition(Country::hinting()->fieldName()->id, -1); // no such records so model will be empty
```
Comment (DarkSide666) [inline demos/collection/lister-ipp.php:53]: Why not `(new Country())->id_field)` ?

Comment (mvorisek) [reply demos/collection/lister-ipp.php:53]: also possible, but this seems better to me when we expect hintable `id` is id field

Comment (mvorisek) [reply demos/collection/lister-ipp.php:53]: best would be `addCondition(false)`, right? :)

/cc @georgehristov

File: demos/init-db.php [hunk 2]
```diff
@@ -188,6 +196,7 @@ protected function init(): void
 
         $this->hasOne($this->fieldName()->client_country_iso, [
             'model' => [Country::class],
+            'our_field' => $this->fieldName()->client_country_iso,
```
Comment (mvorisek) [inline demos/init-db.php]: When prefixing is used, guessing `our_field` for hasOne doesn't work

code: https://github.com/atk4/data/blob/68aa5efd536820d1d7b92061f4c18786425c77e9/src/Reference/HasOne.php#L157

is it possible to fix that?

File: demos/collection/lister-ipp.php (hunk 1 above)
Comment (DarkSide666) [reply demos/collection/lister-ipp.php:53]: Or `addCondition(1, '<>', 1)` :)

**Review by DarkSide666 — APPROVED** [review]

File: demos/init-db.php (hunk 2 above)
Comment (DarkSide666) [reply demos/init-db.php]: Looks like it's working now

**Pull request merged**